From f1b6cf49d43ada8c10dd7d3dc86316d722149db6 Mon Sep 17 00:00:00 2001
From: Marek Kasik <mkasik@redhat.com>
Date: Fri, 17 Jul 2015 10:58:59 +0200
Subject: [PATCH 2/2] notifications: Don't crash because of wrong pointer

Initialize 'new_app_ids' pointer in children_changed()
and check whether it was filled before using it.

https://bugzilla.gnome.org/show_bug.cgi?id=752525
---
 panels/notifications/cc-notifications-panel.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/panels/notifications/cc-notifications-panel.c b/panels/notifications/cc-notifications-panel.c
index 630fb99..91a9317 100644
--- a/panels/notifications/cc-notifications-panel.c
+++ b/panels/notifications/cc-notifications-panel.c
@@ -505,14 +505,17 @@ children_changed (GSettings            *settings,
                   CcNotificationsPanel *panel)
 {
   int i;
-  gchar **new_app_ids;
+  gchar **new_app_ids = NULL;
 
   g_settings_get (panel->master_settings,
                   "application-children",
                   "^as", &new_app_ids);
-  for (i = 0; new_app_ids[i]; i++)
-    maybe_add_app_id (panel, new_app_ids[i]);
-  g_strfreev (new_app_ids);
+  if (new_app_ids != NULL)
+    {
+      for (i = 0; new_app_ids[i]; i++)
+        maybe_add_app_id (panel, new_app_ids[i]);
+      g_strfreev (new_app_ids);
+    }
 }
 
 static void
-- 
2.4.3

